### PR TITLE
improvement(sdcm.argus_test_run): Parse JOB_NAME instead of branch names

### PR DESCRIPTION
**Pull request by k0machi**
This commit changes the way argus will track SCT runs. Previously, it
would use a list of whitelisted SCM branches and organize it this way:
branch, module name, test name (from JOB_NAME url). Now, it will refuse
to run locally (any time get_job_name returns "local_run"), but for any
build system run it will submit the runs into the database - by first
parsing the JOB_NAME environment provided by Jenkins into three
components. This function will also handle certain corner cases (like
releng-testing folder for example), and arbitrary nesting.

[Trello](https://trello.com/c/aDyI6n3Y/4257-add-group-and-test-name-to-all-test-yamls)

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~New configuration option are added and documented (in `sdcm/sct_config.py`)~
- [x] ~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~
- [x] All new and existing unit tests passed (CI)
- [x] ~I have updated the Readme/doc folder accordingly (if needed)~

### CODE REVIEW

File: sdcm/tester.py [hunk 1]
```diff
@@ -372,12 +372,12 @@ def __init__(self, *args):  # pylint: disable=too-many-statements,too-many-local
...
         try:
             argus_test_run = ArgusTestRun.from_sct_config(test_id=UUID(self.test_id),
-                                                          test_module_path=self.test_config.test_name(),
                                                           sct_config=self.params)
             argus_test_run.save()
             return argus_test_run
-        except Exception:  # pylint: disable=broad-except
-            self.log.error("ERROR SETTING UP ARGUS CONNECTION", exc_info=True)
+        except Exception as exc:  # pylint: disable=broad-except
+            self.log.warning("ERROR SETTING UP ARGUS CONNECTION: %s", exc.args[0])
```
Comment (bentsi) [inline sdcm/tester.py:379]: unable to setup  argus connection

File: sdcm/argus_test_run.py [hunk 2]
```diff
@@ -170,6 +169,55 @@ def _prepare_docker_resource_setup(sct_config: SCTConfiguration):
...
+    -> [scylla-master, artifacts, artifacts-amazon2-test]
+    manager-master/releng-testing/debian10-sanity-test
+    -> [manager-master, releng-testing, debian10-sanity-test]
+    manager-master/ubuntu20-sanity-test
+    -> [manager-master, manager-tests, ubuntu20-sanity-test]
+    """
+    release = ""
+    group = ""
+    name = ""
+    if "manager" in details[0] and "releng" in details[1]:
```
Comment (bentsi) [inline sdcm/argus_test_run.py:190]: let's not track releng jobs

File: sdcm/argus_test_run.py [hunk 3]
```diff
@@ -170,6 +169,55 @@ def _prepare_docker_resource_setup(sct_config: SCTConfiguration):
...
+    """
+    Parse a jenkins job url into a three element tuple containing
+    release, group, test names. Most runs should fit into this pattern easily
+    and some will need some additional parsing / adjustments
+    Examples:
+    scylla-master/releng-testing/artifacts/artifacts-ami-test
+    -> [scylla-master, releng-testing-artifacts, artifacts-ami-test]
+    scylla-master/artifacts/artifacts-amazon2-test
+    -> [scylla-master, artifacts, artifacts-amazon2-test]
+    manager-master/releng-testing/debian10-sanity-test
```
Comment (bentsi) [inline sdcm/argus_test_run.py:182]: remove releng-testing

File: sdcm/argus_test_run.py [hunk 4]
```diff
@@ -170,6 +169,55 @@ def _prepare_docker_resource_setup(sct_config: SCTConfiguration):
...
+        release, name, *_ = details
+        group = "manager-tests"
+    elif "scylla-operator" in details[0]:
+        _, release, group, *name = details
+        name = "-".join(name)
+    elif "scylla-staging" in details[0]:
+        release, group, *name_split = details
+        name = "-".join(name_split)
+    elif len(details) == 4:
+        # assume subgroup - combine element 2 and 3
```
Comment (bentsi) [inline sdcm/argus_test_run.py:202]: releng -> remove

File: sdcm/tester.py (hunk 1 above)
Comment (k0machi) [reply sdcm/tester.py]: Done

File: sdcm/argus_test_run.py (hunk 2 above)
Comment (k0machi) [reply sdcm/argus_test_run.py]: Done

File: sdcm/argus_test_run.py (hunk 4 above)
Comment (k0machi) [reply sdcm/argus_test_run.py]: Done

File: sdcm/argus_test_run.py [hunk 5]
```diff
@@ -170,6 +169,47 @@ def _prepare_docker_resource_setup(sct_config: SCTConfiguration):
...
+        release, group, *name_split = details
+        name = "-".join(name_split)
+    elif len(details) == 3:
+        release, group, name = details
+    elif len(details) < 3:
+        LOGGER.warning("Non-parseable build name detected, this run won't be stored in argus: %s", details)
+    else:
+        # try to make up a test name from arbitrary nesting
+        release, *group, name = details
+        group = "-".join(group)
```
Comment (enaydanov) [inline sdcm/argus_test_run.py:205]: Amazingly good candidate for using `match` operator :) Do we have plans to backport it to pre-3.10 branches?

File: sdcm/argus_test_run.py [hunk 6]
```diff
@@ -211,26 +244,29 @@ def warn_once(cls, message: str, *args: list):
...
-        test_group, *_ = test_module_path.split(".")
-        config_files = sct_config.get("config_files")
         job_name = get_job_name()
+        if job_name == "local_run":
+            raise ArgusTestRunError("Will not track a locally run job")
+
         job_url = get_job_url()
+        release_name, group_name, test_name = parse_run_details(job_name.split("/"))
+        if not release_name:
+            raise ArgusTestRunError(f"Unable to track a job: {job_name}", job_name)
```
Comment (enaydanov) [inline sdcm/argus_test_run.py:260]: Is it intended to pass `job_name` as a second argument?

File: sdcm/argus_test_run.py [hunk 7]
```diff
@@ -170,6 +169,47 @@ def _prepare_docker_resource_setup(sct_config: SCTConfiguration):
     return cloud_setup
 
 
+def parse_run_details(details: list) -> tuple:
+    """
+    Parse a jenkins job url into a three element tuple containing
+    release, group, test names. Most runs should fit into this pattern easily
+    and some will need some additional parsing / adjustments
+    Examples:
+    scylla-master/releng-testing/artifacts/artifacts-ami-test
```
Comment (bentsi) [inline sdcm/argus_test_run.py:178]: remove releng-testing example

File: sdcm/argus_test_run.py (hunk 6 above)
Comment (bentsi) [reply sdcm/argus_test_run.py:258]: yes
